apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  managementApiPrefix,
+  workflowLocationKey,
+  workflowResourceGroupNameKey,
+  workflowSubscriptionIdKey,
+  localSettingsFileName,
+} from '../../../constants';
+import { ext } from '../../../extensionVariables';
+import { getLocalSettingsJson } from '../../utils/appSettings/localSettings';
+import { addLocalFuncTelemetry } from '../../utils/funcCoreTools/funcVersion';
+import type { IAzureScriptWizard } from './azureScriptWizard';
+import { createAzureWizard } from './azureScriptWizard';
+import { FileManagement, UserInput } from './iacGestureHelperFunctions';
+import type { IActionContext } from '@microsoft/vscode-azext-utils';
+import type { ILocalSettingsJson } from '@microsoft/vscode-extension';
+import * as AdmZip from 'adm-zip';
+import * as path from 'path';
+import * as portfinder from 'portfinder';
+import * as requestP from 'request-promise';
+import * as vscode from 'vscode';
+
+export async function generateDeploymentScripts(context: IActionContext, folder: vscode.Uri): Promise<void> {
+  try {
+    addLocalFuncTelemetry(context);
+
+    // Setup script context
+    const scriptContext = await setupScriptContext(context, folder);
+
+    // Gather and validate inputs
+    const inputs = await gatherAndValidateInputs(scriptContext, folder);
+
+    // Get source control path
+    const sourceControlPath = await getSourceControlPath(scriptContext);
+
+    // Call API to get deployment artifact
+    const zipContent = await callApiForDeploymentArtifact(inputs);
+
+    // Handle API response
+    await handleApiResponse(zipContent, sourceControlPath, scriptContext);
+  } catch (error) {
+    console.error('An error occurred:', error);
```

Comment:
Should we do something like `vscode.window.showWarningMessage` instead of simply just console log it? All of the other console.errors are ok since they all will end up in this catch scenario. Is there any way we can store the error in telemetry?